libs/frontend/feature-build/src/components/component-menu/ComponentChildStyles.vue

```diff
@@ -86,14 +112,39 @@ const props = defineProps<{
 }>()
 const { component } = toRefs(props)
 
-const selector = ref()
+const getDefaultSelector = (): string | undefined => {
+  return Object.keys(component.value.style.overrides ?? {})[0]
+}
+
+const selector = ref(getDefaultSelector())
 const collapsed = ref(true)
 const editStyleProp = ref<string | undefined>()
 
+const childrenIdSet = computed(
+  () => new Set(component.value.children?.map((child) => child.id)),
+)
+
+const overriddenChildrenIdSet = computed(
+  () => new Set(Object.keys(component.value.style.overrides ?? {})),
+)
+
 const selectors = computed(() => {
-  return (component.value.children ?? []).map((c) => c.id)
+  const allIds = [...childrenIdSet.value, ...overriddenChildrenIdSet.value]
+  const uniqueIds = new Set(allIds)
+  return Array.from(uniqueIds)
+})
+
+const hasMissingSelector = computed(() => {
+  const overriddenChildrenIds = Array.from(overriddenChildrenIdSet.value)
+  return overriddenChildrenIds.some((id) => !childrenIdSet.value.has(id))
 })
 
+const currentSelectorMissing = computed(
+  () =>
+    !!selector.value &&
+    !component.value.children?.some((child) => child.id === selector.value),
+)
+
```

Comment:
The code here seems kind of messy 😬.
My intent was to:

- Be able mark invalid selectors in the menu (selectors where the corresponding children are not there anymore).
- Be able to show the remove button when the selected selector is invalid.
- Use `set.has` instead of `array.includes` to avoid iterating arrays too much, but I'm not sure if it's worth doing it here.

Reply:
It's fine, though I don't think it's makes a big difference here. My one note is that it's probably better to avoid going from Array -> Set -> Array, for example `Object.keys(component.value.style.overrides ?? {})` could be calculated once and used in both `overriddenChildrenIdSet` and `hasMissingSelector`.